src/measure_theory/measure/null_measurable.lean

```diff
@@ -0,0 +1,381 @@
+  null_measurable_set (⋂₀ s) μ :=
+measurable_set.sInter hs h
+
+lemma Inter_Prop {p : Prop} {f : p → set α} (hf : ∀ b, null_measurable_set (f b) μ) :
+  null_measurable_set (⋂ b, f b) μ :=
+measurable_set.Inter_Prop hf
+
+lemma Inter_fintype [fintype ι] {f : ι → set α} (h : ∀ b, null_measurable_set (f b) μ) :
+  null_measurable_set (⋂ b, f b) μ :=
+measurable_set.Inter_fintype h
+
+@[simp] protected lemma union (hs : null_measurable_set s μ) (ht : null_measurable_set t μ) :
+  null_measurable_set (s ∪ t) μ :=
+hs.union ht
+
+protected lemma union_null (hs : null_measurable_set s μ) (ht : μ t = 0) :
+  null_measurable_set (s ∪ t) μ :=
+hs.union (of_null ht)
+
+@[simp] protected lemma inter (hs : null_measurable_set s μ) (ht : null_measurable_set t μ) :
+  null_measurable_set (s ∩ t) μ :=
+hs.inter ht
+
+@[simp] protected lemma diff (hs : null_measurable_set s μ) (ht : null_measurable_set t μ) :
+  null_measurable_set (s \ t) μ :=
+hs.diff ht
+
+@[simp] protected lemma disjointed {f : ℕ → set α} (h : ∀ i, null_measurable_set (f i) μ) (n) :
+  null_measurable_set (disjointed f n) μ :=
+measurable_set.disjointed h n
+
+@[simp] protected lemma const (p : Prop) : null_measurable_set {a : α | p} μ :=
+measurable_set.const p
+
+instance [measurable_singleton_class α] : measurable_singleton_class (null_measurable_space α μ) :=
+⟨λ x, (@measurable_set_singleton α _ _ x).null_measurable⟩
+
+protected lemma insert [measurable_singleton_class (null_measurable_space α μ)]
+  (hs : null_measurable_set s μ) (a : α) :
+  null_measurable_set (insert a s) μ :=
+hs.insert a
+
+lemma exists_measurable_superset_ae_eq (h : null_measurable_set s μ) :
+  ∃ t ⊇ s, measurable_set t ∧ t =ᵐ[μ] s :=
+begin
+  rcases h with ⟨t, htm, hst⟩,
+  refine ⟨t ∪ to_measurable μ (s \ t), _, htm.union (measurable_set_to_measurable _ _), _⟩,
+  { exact diff_subset_iff.1 (subset_to_measurable _ _) },
+  { have : to_measurable μ (s \ t) =ᵐ[μ] (∅ : set α), by simp [ae_le_set.1 hst.le],
+    simpa only [union_empty] using hst.symm.union this }
+end
+
+lemma to_measurable_ae_eq (h : null_measurable_set s μ) :
+  to_measurable μ s =ᵐ[μ] s :=
+begin
+  rw [to_measurable, dif_pos],
+  exact h.exists_measurable_superset_ae_eq.some_spec.snd.2
+end
+
+lemma compl_to_measurable_compl_ae_eq (h : null_measurable_set s μ) :
+  (to_measurable μ sᶜ)ᶜ =ᵐ[μ] s :=
+by simpa only [compl_compl] using h.compl.to_measurable_ae_eq.compl
+
+lemma exists_measurable_subset_ae_eq (h : null_measurable_set s μ) :
+  ∃ t ⊆ s, measurable_set t ∧ t =ᵐ[μ] s :=
+⟨(to_measurable μ sᶜ)ᶜ, compl_subset_comm.2 $ subset_to_measurable _ _,
+  (measurable_set_to_measurable _ _).compl, h.compl_to_measurable_compl_ae_eq⟩
+
+end null_measurable_set
+
+lemma measure_Union {m0 : measurable_space α} {μ : measure α} [encodable ι] {f : ι → set α}
+  (hn : pairwise (disjoint on f)) (h : ∀ i, measurable_set (f i)) :
+  μ (⋃ i, f i) = ∑' i, μ (f i) :=
```

Comment:
is that really the right place for this lemma? Also, can you replace `pairwise (disjoint on f)` with `pairwise_disjoint f` (here and below)?

Reply:
I can move it to `_def`. I need the version for null measurable sets to define completion, so I can't postpone it till the main `measure_space` file.

Reply:
I failed to find `pairwise_disjoint` for indexed families of sets, only `set.pairwise_disjoint`.